crates/containerd-shim-wasm/src/sandbox/containerd/client.rs

```diff
@@ -38,124 +37,117 @@ static MAX_WRITE_CHUNK_SIZE_BYTES: i64 = 1024 * 1024 * 15;
-        self.rt.block_on(async {
-            let mut lease_labels = HashMap::new();
-            // Unwrap is safe here since 24 hours is a valid time
-            let expire = chrono::Utc::now() + chrono::Duration::try_hours(24).unwrap();
-            lease_labels.insert("containerd.io/gc.expire".to_string(), expire.to_rfc3339());
-            let lease_request = containerd_client::services::v1::CreateRequest {
-                id: reference.clone(),
-                labels: lease_labels,
-            };
-
-            let mut leases_client = LeasesClient::new(self.inner.clone());
-
-            let lease = leases_client
-                .create(with_namespace!(lease_request, self.namespace))
-                .await
-                .map_err(|e| ShimError::Containerd(e.to_string()))?
-                .into_inner()
-                .lease
-                .ok_or_else(|| {
-                    ShimError::Containerd(format!("unable to create lease for  {}", reference))
-                })?;
+    async fn lease(&self, reference: String) -> Result<LeaseGuard> {
+        let mut lease_labels = HashMap::new();
+        // Unwrap is safe here since 24 hours is a valid time
+        let expire = chrono::Utc::now() + chrono::Duration::try_hours(24).unwrap();
+        lease_labels.insert("containerd.io/gc.expire".to_string(), expire.to_rfc3339());
+        let lease_request = containerd_client::services::v1::CreateRequest {
+            id: reference.clone(),
+            labels: lease_labels,
+        };
 
-            Ok(LeaseGuard {
-                lease_id: lease.id,
-                address: self.address.clone(),
-                namespace: self.namespace.clone(),
-            })
-        })
+        let mut leases_client = LeasesClient::new(self.inner.clone());
+        let lease = leases_client
+            .create(with_namespace!(lease_request, self.namespace))
+            .await
+            .map_err(|e| ShimError::Containerd(e.to_string()))?
+            .into_inner()
+            .lease
+            .ok_or_else(|| {
+                ShimError::Containerd(format!("unable to create lease for  {}", reference))
+            })?;
+
+        Ok(LeaseGuard::new(
+            leases_client.clone(),
+            lease.id,
+            self.namespace.clone(),
+        ))
     }
 
     #[cfg_attr(feature = "tracing", tracing::instrument(parent = tracing::Span::current(), skip_all, level = "Info"))]
-    fn save_content(
+    async fn save_content(
         &self,
         data: Vec<u8>,
         unique_id: &str,
         labels: HashMap<String, String>,
     ) -> Result<WriteContent> {
         let expected = format!("sha256:{}", digest(data.clone()));
         let reference = format!("precompile-{}", unique_id);
-        let lease = self.lease(reference.clone())?;
+        let lease = self.lease(reference.clone()).await?;
 
-        let digest = self.rt.block_on(async {
+        let digest = 'digest: {
```

Comment:
Oh interesting, I've never seen this rust usage before!!

Reply:
hehe, named blocks, a rather infrequent feature indeed.
wasm also had them, well, rather numbered than named :-)